srcs/Socket.cpp

```diff
@@ -62,7 +65,38 @@ void	Socket::printConnection(void) {
 	std::cout << "server received connection from: " << s << std::endl;
 }
 
-
+void	Socket::initListenSocket(void) {
+	struct addrinfo	hints;
+	struct addrinfo	*ai;
+	struct addrinfo *p;
+
+	// Get a socket for us and bind it
+	memset(&hints, 0, sizeof(hints)); // change to ft_memset later?
+	hints.ai_family = AF_INET;
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_flags = AI_PASSIVE;
+	if (getaddrinfo(NULL, PORT, &hints, &ai) != 0)
+		throw SocketInitException();
+	for (p = ai; p != NULL; p = p->ai_next)
+	{
+		try {
+			createSocket(p->ai_family, p->ai_socktype, p->ai_protocol);
+		} catch (std::exception& e) {
+			continue ;
+		}
+		try {
+			bindSocket(p->ai_addr, p->ai_addrlen);
+		} catch (std::exception& e) {
```

Comment:
I would write the error to ease the debug if something bad occurs

Reply:
Same here:

```
void	Socket::bindSocket(struct sockaddr* ai_addr, socklen_t ai_addrlen) {
	if (bind(sockfd, ai_addr, ai_addrlen) < 0)
		throw Socket::SocketBindException();
	std::cout << "bindSocket worked successfully" << std::endl;
}
```